plugins/aws-catalog-backend/src/BaseEntityProvider.test.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright SeatGeek
+ * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
+ */
+import { getVoidLogger } from '@backstage/backend-common';
+import { TaskInvocationDefinition, TaskRunner } from '@backstage/backend-tasks';
+import { ResourceEntity } from '@backstage/catalog-model';
+import { EntityProviderConnection } from '@backstage/plugin-catalog-node';
+import {
+  BaseEntityProvider,
+  InstanceFilter,
+  InstanceTransformer,
+} from './BaseEntityProvider';
+
+class PersistingTaskRunner implements TaskRunner {
+  private tasks: TaskInvocationDefinition[] = [];
+
+  getTasks() {
+    return this.tasks;
+  }
+
+  run(task: TaskInvocationDefinition): Promise<void> {
+    this.tasks.push(task);
+    return Promise.resolve(undefined);
+  }
+}
+
+const logger = getVoidLogger();
+
+interface SomeInstance {
+  name: string;
+}
+
+const simpleTransformer: InstanceTransformer<SomeInstance> = (
+  i: SomeInstance,
+): Promise<ResourceEntity> => {
+  return Promise.resolve({
+    apiVersion: 'backstage.io/v1alpha1',
+    kind: 'Resource',
+    metadata: {
+      name: i.name,
+    },
+    spec: {
+      type: 'test',
+      owner: 'somebody',
+    },
+  });
+};
+
+class TestableEntityProvider extends BaseEntityProvider<SomeInstance> {
+  constructor(
+    id: string,
+    taskRunner: TaskRunner,
+    filter?: InstanceFilter<SomeInstance> | undefined,
+  ) {
+    super({
+      id,
+      logger,
+      filter,
+      taskRunner,
+      transformer: simpleTransformer,
+    });
+  }
+
+  protected getArn(resource: SomeInstance): string {
+    return `arn:aws:whatever:us-east-1:0123456789:${resource.name}`;
+  }
+
+  protected getInstances(): Promise<SomeInstance[]> {
+    return Promise.resolve([{ name: 'foo' }, { name: 'bar' }, { name: 'baz' }]);
+  }
```

Comment:
same nit here, to take or leave.

```suggestion
 async protected getInstances(): Promise<SomeInstance[]> {
    return [{ name: 'foo' }, { name: 'bar' }, { name: 'baz' }];
  }
```

Reply:
Also fixed via abfd75c